utils/envManagementUtils_test.go

```diff
@@ -0,0 +1,202 @@
+	expectedEndpoint := getSampleEndpoints().Environments[devName].RegistrationEndpoint
+	if returnedEndpoint != expectedEndpoint {
+		t.Errorf("Expected '%s', got '%s'\n", expectedEndpoint, returnedEndpoint)
+	}
+	defer os.Remove(testEndpointsFilePath)
+
+}
+
+func TestGetClientIDOfEnv(t *testing.T) {
+	writeCorrectKeys()
+
+	returnedKey := GetClientIDOfEnv(devName, testKeysFilePath)
+	expectedKey := getSampleKeys().Environments[devName].ClientID
+	if returnedKey != expectedKey {
+		t.Errorf("Expected '%s', got '%s'\n", expectedKey, returnedKey)
+	}
+	defer os.Remove(testKeysFilePath)
+
+}
+
+func TestGetClientSecretOfEnv(t *testing.T) {
+	writeCorrectKeys()
+
+	returnedKey := GetClientSecretOfEnv(devName, devPassword, testKeysFilePath)
+	expectedKey := Decrypt([]byte(GetMD5Hash(devPassword)), getSampleKeys().Environments[devName].ClientSecret)
+
+	if returnedKey != expectedKey {
+		t.Errorf("Expected '%s', got '%s'\n", expectedKey, returnedKey)
+	}
+	defer os.Remove(testKeysFilePath)
+}
+
+func TestGetUsernameOfEnv(t *testing.T) {
+	writeCorrectKeys()
+
+	returnedKey := GetUsernameOfEnv(devName, testKeysFilePath)
+	expectedKey := getSampleKeys().Environments[devName].Username
+
+	if returnedKey != expectedKey {
+		t.Errorf("Expected '%s', got '%s'\n", expectedKey, returnedKey)
+	}
+	defer os.Remove(testKeysFilePath)
+}
+
+func TestAddNewEnvToKeysFile1(t *testing.T) {
+	writeCorrectKeys()
+	var envKeys EnvKeys = EnvKeys{"staging_username", "staging_client_id", "staging_client_secret"}
+
+	AddNewEnvToKeysFile("staging", envKeys, testKeysFilePath)
+	defer os.Remove(testKeysFilePath)
+}
+
+func TestAddNewEnvToKeysFile2(t *testing.T) {
+	var envKeys EnvKeys = EnvKeys{"staging_username", "staging_client_id", "staging_client_secret"}
+
+	AddNewEnvToKeysFile("staging", envKeys, testKeysFilePath)
+	defer os.Remove(testKeysFilePath)
+}
+
+// Case 1: Correct Details
+func TestRemoveEnvFromKeysFile1(t *testing.T) {
+	writeCorrectEndpoints()
+	writeCorrectKeys()
+	err := RemoveEnvFromKeysFile(devName, testKeysFilePath, testEndpointsFilePath)
+	if err != nil {
+		t.Error("Error removing env from keys file: " + err.Error())
+	}
+
+	defer removeFiles()
+}
+
+// Case 2: Env not available in keys file
+func TestRemoveEnvFromKeysFile2(t *testing.T) {
+	writeCorrectEndpoints()
+
+	// write incorrect keys
+	envKeysAll.Environments = make(map[string]EnvKeys)
+	qaEncryptedClientSecret := Encrypt([]byte(GetMD5Hash(qaPassword)), "qa_client_secret")
+	envKeysAll.Environments[qaName] = EnvKeys{"qa_client_id", qaEncryptedClientSecret, qaUsername}
+	WriteConfigFile(envKeysAll, testKeysFilePath)
+	// end of writing incorrect keys
+
+	err := RemoveEnvFromKeysFile(devName, testKeysFilePath, testEndpointsFilePath)
+	if err == nil {
+		t.Error("No error returned. 'Env not found in keys file' error expected")
+	}
+
+	defer removeFiles()
+}
+
+// Case 4: Incorrect Endpoints
+func TestRemoveEnvFromKeysFile4(t *testing.T) {
```

Comment:
We usually start method names with simple letters when writing Java code. Is there a convention in Go to start with caps? If not, let's stick to the norm.

Reply:
In Go, names of functions starting with uppercase letters are accessible across packages while those that start with lowercase letters are limited to a package.